src/abstractdataframe/io.jl

```diff
@@ -92,16 +99,74 @@ function html_escape(cell::AbstractString)
     return cell
 end
 
-Base.show(io::IO, mime::MIME"text/html", df::AbstractDataFrame; summary::Bool=true) =
-    _show(io, mime, df, summary=summary)
+"""
+    show(io::IO, mime::MIME, df::AbstractDataFrame;
+         allrows::Bool = !get(io, :limit, true),
+         allcols::Bool = !get(io, :limit, true),
+         summary::Bool=true)
+
+Render a data frame to an I/O stream using `mime`.
+
+# Arguments
+- `io::IO`: The I/O stream to which `df` will be printed.
+- `mime::MIME`: supoorted mime types are: `"text/html"`, `"text/latex"`, `"text/csv"`,
+   `"text/tab-separated-values"`
+- `df::AbstractDataFrame`: The data frame to print.
+- `allrows::Bool `: supported only for HTML and LaTeX mime types;
+   whether to print all rows, rather than a default number defined in
+   `HTML_NROWS` and `LATEX_NROWS` constants. The default value can be overriden by
+   `IOContext` property `nrows`.
+- `allcols::Bool`: supported only for HTML and LaTeX mime types;
+   whether to print all columns, rather than a default number defined in
+   `HTML_NCOLS` and `LATEX_NCOLS` constants. The default value can be overriden by
+   `IOContext` property `ncols`.
+- `summary::Bool = true`: supported only for HTML mime type;
+   Whether to print a brief string summary of the data frame.
+
+# Examples
+```jldoctest
+julia> show(stdout, MIME("text/latex"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+\\begin{tabular}{r|cc}
+        & A & B\\\\
+        \\hline
+        & Int64 & String\\\\
+        \\hline
+        1 & 1 & x \\\\
+        2 & 2 & y \\\\
+        3 & 3 & z \\\\
+\\end{tabular}
+14
+
+julia> show(stdout, MIME("text/csv"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+"A","B"
+1,"x"
+2,"y"
+3,"z"
+```
+"""
+Base.show(io::IO, mime::MIME"text/html", df::AbstractDataFrame;
+          allrows::Bool = !get(io, :limit, true),
```

Comment:
Shouldn't we default to `false`? Otherwise we're going to break the use case where you want to print all the data to a file. If Jupyter doesn't set `:limit=>true`, that should be fixed.

Reply:
I thought we should not print all rows by default (the same with columns). The idea is that if you want to print all rows/columns you use `show` and explicitly set `allrows=true` or `allcols=true`. I think it should be not a huge burden. I would assume that users will almost never need to print whole a `DataFrame` to a file in e.g. `text/html` format so if you want to do it it is not a big burden to wrap `df` in `show` with appropriate kwargs.

Reply:
I'm not sure I agree:
- A data frame can be fairly small, notably if it contains summary statistics over groups.
- For CSV output we definitely want to print everything, so better be consistent for HTML. Also the text/plain method defaults to `false`.
- I don't see in what situations one would like to print a partial data frame as HTML or LaTeX to a file (as opposed to Jupyter, which sets `:limit=>true`).

Reply:
OK